js-intermediate.js

```diff
@@ -14,7 +35,28 @@ var fullArr1 = [4, 9, 0, "7", 8, true, "hey", 7, 199, -9, false, "hola"]
 var fullArr2 = ["hello", 7, 23, -823, false, 78, null, "67", 6, "number"]
 // Expected output: [7, 23, -823]
 
-
+// terenary requires an else... so tried to break out the interior function not working yet
+// const oddChecker1 = (arg) => {
+//   let newArray = arg.map(element => 
+//       if (typeof element === "number" && element%2 !== 0){
+//         element
+//       })
+//   return newArray
+// }
+// console.log(`#2A: ${oddChecker1(fullArr1)}`)
+
+const oddChecker = (arg) => {
+  let newArray = []
+  for(let i = 0; i <arg.length; i++){
+    if(typeof arg[i] === "number"){
+      if(!arg[i]%2 === 0){
+        newArray.push(arg[i])
+      }
+    }
+  }
+  return newArray
+}
+console.log(`#2: ${oddChecker(fullArr1)}`)
 
```

Comment:
If you move the bang operator from in front of the arg to using it in the evaluation you get the outcome you expect.
arg[i[%2 !== 0